api/web3server.go

```diff
@@ -570,14 +568,14 @@ func (svr *web3Handler) getBlockTransactionCountByNumber(in *gjson.Result) (inte
 	if err != nil {
 		return nil, err
 	}
-	blkMetas, err := svr.coreService.BlockMetas(num, 1)
+	res, err := svr.coreService.BlockByHeight(num)
 	if err != nil {
+		if errors.Cause(err) == ErrNotFound {
+			return nil, nil
```

Comment:
it returns `errInvalidBlock` in current code? see left

Reply:
err is handled differently in `BlockByHeight()`